### ISSUE TITLE
It uses the description instead of the name so TestParams will not work because of duplicate names

### ISSUE THREAD

**Issue by air2** [TylerLeonhardt/vscode-pester-test-adapter#13]
```
It "Some test <ExpectedPath>" -TestCases @(@{ IsAdmin = $true; ExpectedPath = 'cert:\LocalMachine\My' },@{ IsAdmin = $false; ExpectedPath = 'cert:\CurrentUser\My' }) 
{
...
}
```
This result in two tests with the same name: **Some Test** because the description in **TestExplorerResults.xml** is **Some tests** where the name is **Some test cert:\LocalMachine\My** and **Some  test cert:\CurrentUser\My**

See the xml:
```
<test-case description="Some test" name="Some test.To run all the test cases.Some test(&quot;cert:\LocalMachine\My&quot;,True)" time="0.234" asserts="0" success="False" result="Failure" executed="True">
...
</test-case>
<test-case description="Some test" name="Some test.To run all the test cases.Some test(&quot;cert:\CurrentUser\My&quot;,False)" time="0.1041" asserts="0" success="False" result="Failure" executed="True">
....
</test-case>
```

Can this be configured (For example to let me choose it uses the name field instead of the description.)

**Comment by TylerLeonhardt** [comment]
Good idea

**Comment by TylerLeonhardt** [comment]
I've thought about it some more... @nohwnd is it possible for Pester to include the `(&quot;cert:\CurrentUser\My&quot;,False)` in the description?

**Comment by TylerLeonhardt** [comment]
Perhaps the `JUnitXml` output is easier to manage? I'm curious about this.

**Comment by nohwnd** [comment]
Yeah but I think we had problems with that in past because the description was too long when a data set was provided that serialized into a long text.

**Comment by TylerLeonhardt** [comment]
@nohwnd any idea if JUnitXml is any better as a format to manage? Does it include any additional details that the XUnitXml one doesn't?

**Comment by PsychoData** [comment]
I'm running into this myself in the output when I try to use Pester tests like 
```posh
It "Check <ParamName> File" { } -TestCases $ParamsHashTable
```

It looks like the Extension is using the line `<test-suite type="ParameterizedTest"`  and the `description` attribute of it. 
![image](https://user-images.githubusercontent.com/3719116/103329105-48c64500-4a29-11eb-9b9a-4227d5f67bfb.png)


Based on the testing I tried, it only seems to be a problem when there are TestCases. 
Maybe when the xmlNode type is ParameterizedTest it could just drill down and add that nodes' results? 

I know hardly anything about typescript, but if I have the debugger on a break point at this line https://github.com/TylerLeonhardt/vscode-pester-test-adapter/blob/56165af3eff094188f46611ceca08ce03bc2d2a9/src/pesterTests.ts#L272

and run this 
```xmlChild.results["test-suite"][xmlChild.results["test-suite"].length-2].results['test-case']._attributes.description```

basically, if that label resolves to this, it looks like it is returning the correct test name - but I'm not certain how to be certain that those are the complete tests
`xmlChild.results["test-suite"][<LastIndex>].results['test-case']._attributes.description`

You can pull my data and tests from [this branch](https://github.com/PsychoData/msportals.io/tree/feature/workflow-PesterTests) if you want - as long as you promise not to judge my probably terrible Pester tests - I'm just learning it

**Comment by PsychoData** [comment]
actually, maybe not as important cause it doesn't have the actual tests or TestCases in it, but it looks like the dynamic doesn't work for Contexts either, possibly? 

![image](https://user-images.githubusercontent.com/3719116/103330629-60ed9280-4a30-11eb-8b20-d30cefec71c6.png)

though it does work on the output of the tests 
![image](https://user-images.githubusercontent.com/3719116/103330671-8aa6b980-4a30-11eb-8861-d61339c7e79a.png)

But, hey, if it worked on TestCases that would be a great improvement 😄

**Comment by TylerLeonhardt** [comment]
Ok so I'm going to call this issue fixed by #29... the problem is, when we discover what tests are in a file, we call into Pester APIs here:
https://github.com/TylerLeonhardt/vscode-pester-test-adapter/blob/master/src/powershellScripts.ts#L10-L71

Pester only gives us the Path with the `Some test <ExpectedPath>` and _not_ the actual resolved value `Some test C:\blah\blah`

Since this is the case, I can't "know" what the actual test name is until you run the test... additionally, Pester doesn't give the ability to run a single test case of an `It` AFAIK so having each test case show up in the tree view of VS Code doesn't make sense.

**Comment by TylerLeonhardt** [comment]
I wish the 'dynamic' value was given when we look for tests so that I could do something clever (or at least show the resolved name for `Context`'s!) but I'll defer that to @nohwnd for his expert advice. For now, I'll say this is fixed as best as I can

**Comment by PsychoData** [comment]
nah, you can run it on a single object, you just have to force it to pass an array to -TestCases  even when there is only 1 item in the array to pass. 

I'm doing this with the `-TestCases @($_) ` - the `@( $variable )` will make an array of one object, or any other objects inside it

![image](https://user-images.githubusercontent.com/3719116/103433099-2a8f4f00-4bb9-11eb-8f2f-4272db4f5504.png)

![image](https://user-images.githubusercontent.com/3719116/103433151-0ed87880-4bba-11eb-887d-192c58024bdb.png)

**Comment by PsychoData** [comment]
Yeah - I don't know if there is much you could do to find the names during discovery, but if it updated the names as the tests ran, after discovery had already been done before, then the output of the test shows the full, resolved names

**Comment by TylerLeonhardt** [comment]
> nah, you can run it on a single object, you just have to force it to pass an array to -TestCases even when there is only 1 item in the array to pass.

I meant, without modifying the script. If your script has an array of TestCases, and I want to run only the 3rd test case, how would I do that without modifying the script?

> but if it updated the names as the tests ran

Yeah that's worth investigating. That should be a separate issue.

**Comment by PsychoData** [comment]
that is true - currently when I try to re-run tests for the It with TestCases, it reruns all of the test cases for that test 
that's still better than rerunning all of the whole suite of tests, though. 

Only other thing I could think of would be to reference it by path to match the name, though it would be kind of bulky. 

Or if there was some sort of test ID number, but I don't see a way to filter down which tests would run without modifying the test scripts without it. 
And you couldn't even say to run Check/It block 14 or whichever with TestCase 19, because that list of TestCases could potentially have changed too

**Comment by TylerLeonhardt** [comment]
Right. I think allowing you to just run a single test case is out of my hands for now. Work would need to be done in Pester to support it.

**Comment by nohwnd** [comment]
> I wish the 'dynamic' value was given when we look for tests so that I could do something clever (or at least show the resolved name for `Context`'s!) but I'll defer that to @nohwnd Jakub Jares FTE for his expert advice. For now, I'll say this is fixed as best as I can

I think that is really hard for you to do, because the <> templates are resolved based on variables that are present when the test is running. You can partially populate it from the Data on the test (+ all it's parent blocks), but you won't be able to do it fully if user uses variables that are coming from BeforeAll, BeforeEach. 

> Or if there was some sort of test ID number, but I don't see a way to filter down which tests would run without modifying the test scripts without it.

Pester does not have api for this, but if this extension detects changes in the file and rediscovers it (and updates the tree of tests), then passing test case id by index should work just fine in most cases. I can't think of a more robust way: We can't compare instances by reference because we don't share them. We can't compare text / ast because there is nothing preventing test cases from being the same in the same collection. So index looks like a simplest option if you stay updated. 

If the discovered is AST based then imho line + column of the test case is the most bulletproof, and aligns with how filtering for running single test works. 

This won't work for test cases that are generated by a function, but that's imho okay.
